qiskit/algorithms/minimum_eigensolvers/adapt_vqe.py

```diff
@@ -0,0 +1,312 @@
+
+        prev_op_indices: list[int] = []
+        theta: list[float] = []
+        max_grad: tuple[float, Optional[PauliSumOp]] = (0.0, None)
+        self._excitation_list = []
+        iteration = 0
+        while self.max_iterations is None or iteration < self.max_iterations:
+            iteration += 1
+            logger.info("--- Iteration #%s ---", str(iteration))
+            # compute gradients
+            cur_grads = self._compute_gradients(theta, operator)
+            # pick maximum gradient
+            max_grad_index, max_grad = max(
+                enumerate(cur_grads), key=lambda item: np.abs(item[1][0])
+            )
+            # store maximum gradient's index for cycle detection
+            prev_op_indices.append(max_grad_index)
+            # log gradients
+            if np.abs(max_grad[0]) < self.threshold:
+                if iteration == 1:
+                    raise QiskitError(
+                        "Gradient choice is not suited as it leads to all zero gradients gradients. "
+                        "Try a different gradient method."
+                    )
+                logger.info(
+                    "AdaptVQE terminated successfully with a final maximum gradient: %s",
+                    str(np.abs(max_grad[0])),
+                )
+                termination_criterion = TerminationCriterion.CONVERGED
+                break
+            # check indices of picked gradients for cycles
+            if self._check_cyclicity(prev_op_indices):
+                logger.info("Final maximum gradient: %s", str(np.abs(max_grad[0])))
+                termination_criterion = TerminationCriterion.CYCLICITY
+                break
+            # add new excitation to self._ansatz
+            self._excitation_list.append(self._excitation_pool[max_grad_index])
+            theta.append(0.0)
+            # run VQE on current Ansatz
+            self._tmp_ansatz.operators = self._excitation_list
+            self.solver.ansatz = self._tmp_ansatz
+            self.solver.initial_point = theta
+            raw_vqe_result = self.solver.compute_minimum_eigenvalue(operator)
+            theta = raw_vqe_result.optimal_point.tolist()
+        else:
+            # reached maximum number of iterations
+            termination_criterion = TerminationCriterion.MAXIMUM
+            logger.info("Maximum number of iterations reached. Finishing.")
+            logger.info("Final maximum gradient: %s", str(np.abs(max_grad[0])))
+
+        result = AdaptVQEResult()
+        result.combine(raw_vqe_result)
+        result.num_iterations = iteration
+        result.final_max_gradient = max_grad[0]
+        result.termination_criterion = termination_criterion
+
+        # once finished evaluate auxiliary operators if any
+        if aux_operators is not None:
+            bound_ansatz = self.solver.ansatz.bind_parameters(result.optimal_point)
+
+            aux_values = estimate_observables(self.solver.estimator, bound_ansatz, aux_operators)
```

Comment:
```suggestion
            aux_values = estimate_observables(
                self._solver.estimator,
                self._solver.ansatz,
                aux_operators,
                result.optimal_point,
            )
```
`estimate_observables` was rewritten to take a parameterized ansatz to avoid creating a new bound ansatz.